applets/wncklet/wayland-backend.c

```diff
@@ -457,24 +541,173 @@ static ToplevelTask *
 toplevel_task_new (TasklistManager *tasklist, struct zwlr_foreign_toplevel_handle_v1 *toplevel)
 {
 	ToplevelTask *task = g_new0 (ToplevelTask, 1);
+	GtkWidget *button;
+	GtkOrientation orient;
 
+	buttons = buttons + 1;
+	orient = gtk_orientable_get_orientation(GTK_ORIENTABLE(tasklist->outer_box));
 	task->button = gtk_button_new ();
 	g_signal_connect (task->button, "clicked", G_CALLBACK (toplevel_task_handle_clicked), task);
 
-	task->icon = gtk_image_new_from_icon_name ("unknown", GTK_ICON_SIZE_MENU);
+
+	task->icon = gtk_image_new_from_icon_name ("unknown", icon_size);
 
 	task->label = gtk_label_new ("");
 	gtk_label_set_width_chars (GTK_LABEL (task->label), -1);
 	gtk_label_set_ellipsize (GTK_LABEL (task->label), PANGO_ELLIPSIZE_END);
 	gtk_label_set_xalign (GTK_LABEL (task->label), 0.0);
 
 	GtkWidget *box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-	gtk_box_pack_start (GTK_BOX (box), task->icon, FALSE, FALSE, 2);
-	gtk_box_pack_start (GTK_BOX (box), task->label, TRUE, TRUE, 3);
+	gtk_box_pack_start (GTK_BOX (box), task->icon, FALSE, FALSE, 6);
+	gtk_box_pack_start (GTK_BOX (box), task->label, TRUE, TRUE, 2);
+
 	gtk_container_add (GTK_CONTAINER (task->button), box);
 	gtk_widget_set_name (task->button , "tasklist-button");
 	gtk_widget_show_all (task->button);
 
+	/* Buttons on a vertical panel are not affected by how many are needed
+	 * GTK handles compressing contents as needed as the window width tells
+	 * GTK how much space to allocate the label and icon. Buttons will use
+	 * the full width of a vertical panel without any special attention
+	 * so break out here instead of breaking the vertical panel case
+	 */
+
+	if (orient == GTK_ORIENTATION_VERTICAL)
+	{
+		gtk_widget_show_all (task->button);
+		task->toplevel = toplevel;
+		zwlr_foreign_toplevel_handle_v1_add_listener (toplevel,
+							&foreign_toplevel_handle_listener,
+							task);
```

Comment:
indents, file up with some spaces after last tab.
```suggestion
		zwlr_foreign_toplevel_handle_v1_add_listener (toplevel,
							      &foreign_toplevel_handle_listener,
							      task);
```

Reply:
When we get these tabs out, this one will be possible to align perfectly

Reply:
Setting the suggested mixed tabs and spaces here. This will definately need to full file reformat later

Reply:
Ihmo, it's ok to use only tabs in a file + some spaces at the end if the tab doesn't match exactly.
Didn't look at the rest of the file but i think no reformat is needed.